File: v2/backend/app/schemas/support.py
"""Pydantic schemas for Customer Support Tickets."""

from typing import Optional, List
from datetime import datetime
from pydantic import BaseModel, ConfigDict, Field


class TicketReplyItem(BaseModel):
    sender_name: str
    sender_role: str  # "customer", "admin", "agent"
    message: str
    created_at: Optional[str] = None


class SupportTicketCreateRequest(BaseModel):
    category: str = Field(..., description="Booking, Payment, Cancellation, Refund, Account, Service, Other")
    subject: str = Field(..., min_length=3, max_length=255)
    description: str = Field(..., min_length=5)
    booking_id: Optional[str] = Field(None, description="Optional booking reference ID")


class SupportTicketReplyRequest(BaseModel):
    message: str = Field(..., min_length=1, max_length=5000)


class SupportTicketResponse(BaseModel):
    id: str
    ticket_code: str
    user_id: str
    user_name: str
    user_email: str
    booking_id: Optional[str] = None
    category: str
    subject: str
    description: str
    status: str  # OPEN, IN_PROGRESS, RESOLVED, CLOSED
    priority: str  # LOW, MEDIUM, HIGH, URGENT
    responses: List[TicketReplyItem] = Field(default_factory=list)
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None
    resolved_at: Optional[datetime] = None

    model_config = ConfigDict(from_attributes=True)


class SupportTicketListResponse(BaseModel):
    tickets: List[SupportTicketResponse]
    total: int

File: v2/frontend/src/types/support.ts
export interface TicketReply {
  sender_name: string;
  sender_role: "customer" | "admin" | "agent" | string;
  message: string;
  created_at?: string | null;
}

export interface SupportTicket {
  id: string;
  ticket_code: string;
  user_id: string;
  user_name: string;
  user_email: string;
  booking_id?: string | null;
  category: "Booking" | "Payment" | "Cancellation" | "Refund" | "Account" | "Service" | "Other" | string;
  subject: string;
  description: string;
  status: "OPEN" | "IN_PROGRESS" | "RESOLVED" | "CLOSED" | string;
  priority: "LOW" | "MEDIUM" | "HIGH" | "URGENT" | string;
  responses: TicketReply[];
  created_at?: string | null;
  updated_at?: string | null;
  resolved_at?: string | null;
}

export interface SupportTicketListResponseData {
  tickets: SupportTicket[];
  total: number;
}

export interface CreateTicketPayload {
  category: string;
  subject: string;
  description: string;
  booking_id?: string;
}

export interface TicketReplyPayload {
  message: string;
}
